public/js/discover/trie.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable no-unused-vars */
+
+const categories = [
+  'canned food',
+  'produce',
+  'cleaning supplies',
+  'blankets',
+  'clothes',
+  'pet food',
+  'plastic bags',
+  'books',
+  'school supplies',
+  'children\'s toys',
+  'hair products',
+  'deoderant',
+  'soap',
+  'bottled drinks',
+  'computers',
+  'cars',
+]
+
+const categoryTrie = new Trie(categories);
+
+/**
+ * Gives autocomplete suggestions for item categories.
+ * @param {string} input The prefix to base autocomplete suggestions on.
+ * @return {array} The list of autocomplete suggestions.
+ */
+function getAutocompleteCategories(input) {
```

Comment:
```suggestion
window.getAutocompleteCategories = function getAutocompleteCategories(input) {
```

Reply:
this will solve the `no-unused-variables` error and should keep the same functionality.

Reply:
Should I do this even if I don't use `getAutocompleteCategories()` in the HTML file? I was planning on calling it from another js file.

Reply:
yes I would still recommend doing so.